modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/DropdownMenu.js

```diff
@@ -64,28 +64,35 @@ export default function DropdownMenu({
 			/>
 
 			<div className="quick-action-menu">
-				{items.map(({data, href, icon, quickAction, ...rest}) =>
-					data?.action && quickAction ? (
-						<ClayButtonWithIcon
-							className="component-action quick-action-item"
-							displayType="unstyled"
-							small={true}
-							symbol={icon}
-							{...rest}
-						/>
-					) : (
-						href &&
-						icon &&
-						quickAction && (
-							<ClayLink
+				{items.map(
+					(
+						{data, href, icon, label, quickAction, ...rest},
+						index
+					) => {
+						const key = (label || icon || 'item-key').toLowerCase();
+
+						return data?.action && quickAction ? (
+							<ClayButtonWithIcon
 								className="component-action quick-action-item"
-								href={href}
+								displayType="unstyled"
+								key={`${key}-${index}`}
```

Comment:
Using the index in the key is not a good idea because different sortings would break the unicity of the row. Note that I don't know if the data can be returned sorted differently.

But if it can be avoided we should try it :)

Reply:
Yes, I agree about the index. No sorting in this case I think. I believe I can avoid the use of the index as label or icon props should be unique strings in the component, but I'm not sure. Gonna check for an unique combination